test/unit/proxy_test.rb

```diff
@@ -159,6 +169,14 @@ def test_sandbox_endpoint_validation
       proxy.staging_endpoint = ''
 
       assert proxy.valid?
+
+      proxy.sandbox_endpoint = "http://localhost:8080"
+      assert proxy.valid?
+
+      service.deployment_option = 'hosted'
+      service.save
```

Comment:
In order to make sure it is saved well, use `#save!` because `save` just returns `true` or `false` but does not fail

Reply:
Agreed